cli/src/tests/query_test.rs

```diff
@@ -737,6 +737,57 @@ fn test_query_matches_capturing_error_nodes() {
     });
 }
 
+#[test]
+fn test_query_matches_capturing_missing_nodes() {
+    allocations::record(|| {
+        let language = get_language("javascript");
+        let query = Query::new(
+            &language,
+            "
+            (MISSING) @missing-node
+            ",
+        )
+        .unwrap();
+
+        // Missing anonymous nodes
+        assert_query_matches(
+            &language,
+            &query,
+            "if ({a: 'b'} {c: 'd'}) {
+  x = function(a) { b; } function(c) { d; }
+}",
+            &[(0, vec![("missing-node", "")])],
+        );
+
+        let language = get_language("c");
+        let query = Query::new(
+            &language,
+            "(field_expression (identifier) (MISSING) @field-ident)
+            (MISSING) @broad-missing",
+        )
+        .unwrap();
+
+        // Missing named nodes
+        assert_query_matches(
+            &language,
+            &query,
+            "int main() {
+  if (a.) {
+    b();
+    c();
+
+    if (*) d();
+  }
+}",
+            &[
```

Comment:
Unlike the prior comment, this test case is a lot more minimal, let's just comment where the missing nodes are and what is missing, since these missing nodes are ZWTs and we have no clue what's missing.